Client/core/Graphics/CGraphics.cpp

```diff
@@ -824,25 +824,32 @@ void CGraphics::DrawRectQueued(float fX, float fY, float fWidth, float fHeight,
 void CGraphics::DrawCircleQueued(float fX, float fY, float fRadius, float fStartAngle, float fStopAngle, unsigned long ulColor, unsigned long ulColorCenter,
                                  short siSegments, float fRatio, bool bPostGUI)
 {
+    // Check if window is minimized so we don't calculate vertices for no reason.
     if (g_pCore->IsWindowMinimized())
         return;
 
-    // Set up a queue item
-    sDrawQueueItem Item;
-    Item.eType = QUEUE_CIRCLE;
-    Item.blendMode = m_ActiveBlendMode;
-    Item.Circle.fX = fX;
-    Item.Circle.fY = fY;
-    Item.Circle.fRadius = fRadius;
-    Item.Circle.fStartAngle = fStartAngle;
-    Item.Circle.fStopAngle = fStopAngle;
-    Item.Circle.bPostGUI = bPostGUI;
-    Item.Circle.fSegments = siSegments;
-    Item.Circle.fRatio = fRatio;
-    Item.Circle.ulColor = ulColor;
-    Item.Circle.ulColorCenter = ulColorCenter;
-    // Add it to the queue
-    AddQueueItem(Item, bPostGUI);
+    auto pVecVertices = new std::vector<PrimitiveVertice>();
+    fStartAngle = D3DXToRadian(fStartAngle);
+    fStopAngle = D3DXToRadian(fStopAngle);
+    // Calculate each segment angle
+    const float kfSegmentAngle = (fStopAngle - fStartAngle) / (siSegments-1);

```

Comment:
I remember removing `-1` in `(siSegments-1)`, and it worked without any issues. Any reason for adding it back?

Reply:
Instead of floating point based loop, i made it to use exactly the amount of segments specified in dxDrawCircle call, that required adding ```-1``` again